torch/fx/passes/dce.py

```diff
@@ -0,0 +1,13 @@
+import torch
+
+
+def run_dce_pass(m: torch.fx.GraphModule):
+    """
+    Remove all dead code from the graph, based on their number of users.
+    """
+    # Reverse iterate so that when we remove a node, any nodes used as an input
+    # to that node have an updated user count that no longer reflects the
+    # removed node.
+    for node in reversed(m.graph.nodes):
+        if node.op not in {"placeholder", "output"} and len(node.users) == 0:
+            m.graph.erase_node(node)
```

Comment:
This could probably be a method on `Graph`. That way we can also keep the semantics in one place. In particular, I'm thinking about if in the future we have more side-effectful nodes we need to preserve other than placeholders and outputs (e.g. asserts)